tinymce-basics/customizing-the-table-toolbar.html

```diff
@@ -0,0 +1,583 @@
+                                icon: 'table-delete-row',
+                                shortcut: 'Meta+Alt+Delete',
+                                onAction: () => {
+                                    menuActionManager.deleteRow();
+                                }
+                            },
+                            {
+                                type: 'separator',
+                            },
+                            {
+                                type: 'menuitem',
+                                text: 'Cut row',
+                                icon: 'cut-row',
+                                onAction: () => {
+                                    menuActionManager.cutRow();
+                                }
+                            },
+                            {
+                                type: 'menuitem',
+                                text: 'Copy row',
+                                icon: 'duplicate-row',
+                                onAction: () => {
+                                    menuActionManager.copyRow();
+                                }
+                            },
+                            {
+                                type: 'menuitem',
+                                text: 'Paste row before',
+                                icon: 'paste-row-before',
+                                onAction: () => {
+                                    menuActionManager.pasteRowBefore();
+                                },
+                                onSetup: (buttonApi) => {
+                                    // Check the clipboard for any rows and disable the menu item if there is none
+                                    // https://www.tiny.cloud/docs/plugins/table/#api
+                                    // https://www.tiny.cloud/docs/ui-components/typesoftoolbarbuttons/#api
+                                    const state = editor.plugins.table.getClipboardRows().length > 0 ? false : true;
+                                    buttonApi.setDisabled(state);
+                                }
+                            },
+                            {
+                                type: 'menuitem',
+                                text: 'Paste row after',
+                                icon: 'paste-row-after',
+                                onAction: () => {
+                                    menuActionManager.pasteRowAfter();
+                                },
+                                onSetup: (buttonApi) => {
+                                    // Check the clipboard for any rows and disable the menu item if there is none
+                                    // https://www.tiny.cloud/docs/plugins/table/#api
+                                    // https://www.tiny.cloud/docs/ui-components/typesoftoolbarbuttons/#api
+                                    const state = editor.plugins.table.getClipboardRows().length > 0 ? false : true;
```

Comment:
This won't update as needed, is that expected? Given this is only meant to be a basic demo that might be fine. If not, then we probably want to hook in a `NodeChange` handler and update whenever the node changes.

Reply:
I'm not sure I understand. what you mean by "update"? The onSetup is run every time the menu is opened, right? So it simply check the clipboard for rows/columns on every opening and toggles the enabled state as needed, or have I got it wrong?

Reply:
While it's a rare case, the selection can be changed to be inside a table cell while a menu is open. So in that case you need to update the menu item state to reflect the new selection. That's all I meant and the rest of what you said is correct.